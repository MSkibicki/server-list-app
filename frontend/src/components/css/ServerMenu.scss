@import "./Global.scss";

.server-icon:hover {
  cursor: pointer;
}
