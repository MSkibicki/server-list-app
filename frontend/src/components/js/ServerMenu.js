import React, { useState } from "react";
import "../css/ServerMenu.scss";
import MoreHorizIcon from "@material-ui/icons/MoreHoriz";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";

const ServerMenu = ({ serverOn, serverOff, serverStatus }) => {
  const [anchorEl, setAnchorEl] = useState(null);

  const handleClick = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  const handleOnOff = () => {
    handleClose();
    if (serverStatus === "ONLINE") {
      serverOff();
    } else {
      serverOn();
    }
  };

  return (
    <div className="server-icon">
      <MoreHorizIcon
        aria-controls="simple-menu"
        aria-haspopup="true"
        onClick={handleClick}
      ></MoreHorizIcon>
      <Menu
        id="simple-menu"
        anchorEl={anchorEl}
        keepMounted
        open={Boolean(anchorEl)}
        onClose={handleClose}
      >
        <MenuItem onClick={handleOnOff}>
          {`
          ${serverStatus === "ONLINE" ? "Turn Off" : "Turn On"}
          `}
        </MenuItem>
        <MenuItem onClick={handleClose}>Reboot</MenuItem>
      </Menu>
    </div>
  );
};

export default ServerMenu;
