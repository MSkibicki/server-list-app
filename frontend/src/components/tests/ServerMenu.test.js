import ServerMenu from "./../js/ServerMenu";

describe("<ServerMenu />", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<ServerMenu />);

    expect(wrapper).toMatchSnapshot();
  });

  it("has correct div classname", () => {
    const wrapper = shallow(<ServerMenu />);

    const divClass = wrapper.find("div.server-icon").hasClass("server-icon");

    expect(divClass).toBe(true);
  });
});
