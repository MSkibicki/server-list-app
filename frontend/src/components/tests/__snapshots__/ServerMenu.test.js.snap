// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServerMenu /> renders correctly 1`] = `
<div
  className="server-icon"
>
  <MoreHorizIcon
    aria-controls="simple-menu"
    aria-haspopup="true"
    onClick={[Function]}
  />
  <WithStyles(ForwardRef(Menu))
    anchorEl={null}
    id="simple-menu"
    keepMounted={true}
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      
          Turn On
          
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(ForwardRef(Menu))>
</div>
`;
